Sources/StreamChatUI/ChatMessageList/ChatChannelHeaderView.swift

```diff
@@ -0,0 +1,144 @@
+        super.updateContent()
+
+        titleContainerView.content = (titleText, subtitleText)
+
+        makeTimer()
+    }
+
+    /// The title text used to render the title label. By default it is the channel name.
+    open var titleText: String? {
+        guard let channel = channelController?.channel else { return nil }
+        return components.channelNamer(channel, currentUserId)
+    }
+
+    /// The subtitle text used in the subtitle label. By default it shows member online status.
+    open var subtitleText: String? {
+        guard let channel = channelController?.channel else { return nil }
+        guard let currentUserId = self.currentUserId else { return nil }
+
+        if channel.isDirectMessageChannel {
+            guard let member = channel
+                .lastActiveMembers
+                .first(where: { $0.id != currentUserId })
+            else {
+                return nil
+            }
+
+            if member.isOnline {
+                return L10n.Message.Title.online
+            } else if let minutes = member.lastActiveAt
+                .flatMap({ DateComponentsFormatter.minutes.string(from: $0, to: Date()) }) {
+                return L10n.Message.Title.seeMinutesAgo(minutes)
+            } else {
+                return L10n.Message.Title.offline
+            }
+        }
+
+        return L10n.Message.Title.group(channel.memberCount, channel.watcherCount)
+    }
+
+    /// Create the timer to repeatedly update the online status of the members.
+    open func makeTimer() {
+        // Only create the timer if is not created yet and if the interval is not zero.
+        guard timer == nil, statusUpdateInterval > 0 else {
+            return
+        }
+
+        timer = Timer.scheduledTimer(
+            withTimeInterval: statusUpdateInterval,
+            repeats: true
+        ) { [weak self] _ in
+            self?.updateContentIfNeeded()
+        }
+    }
+
+    // MARK: - ChatChannelControllerDelegate Implementation
+
+    open func channelController(
+        _ channelController: _ChatChannelController<ExtraData>,
+        didUpdateChannel channel: EntityChange<_ChatChannel<ExtraData>>
+    ) {
+        switch channel {
+        case .update:
+            updateContentIfNeeded()
+        default:
+            break
+        }
+    }
+
+    open func channelController(
+        _ channelController: _ChatChannelController<ExtraData>,
+        didChangeTypingUsers typingUsers: Set<_ChatUser<ExtraData.User>>
+    ) {
+        // By default the header view is not interested in typing events
+        // but this can be overridden by subclassing this component.
+    }
+
+    open func channelController(
+        _ channelController: _ChatChannelController<ExtraData>,
+        didReceiveMemberEvent: MemberEvent
+    ) {
+        // By default the header view is not interested in member events
+        // but this can be overridden by subclassing this component.
+    }
```

Comment:
we don't have `updateContentIfNeeded` here because for these changes we get `.update` change, right? 🤔

Reply:
Yes 👍